operator/pkg/controllers/etcd/etcd.go

```diff
@@ -169,6 +167,49 @@ func (e *Provider) createSecret(ctx context.Context, controlPlane *v1alpha1.Cont
 	return
 }
 
+func (e *Provider) createStatefulset(ctx context.Context, controlPlane *v1alpha1.ControlPlane) error {
+	statefulSet := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      etcdServiceNameFor(controlPlane.ClusterName()),
+			Namespace: controlPlane.NamespaceName(),
+			OwnerReferences: []metav1.OwnerReference{{
+				APIVersion: controlPlane.APIVersion,
+				Name:       controlPlane.Name,
+				Kind:       controlPlane.Kind,
+				UID:        controlPlane.UID,
+			}},
+		},
+	}
+	result, err := controllerutil.CreateOrPatch(ctx, e.kubeClient, statefulSet, func() (err error) {
+		// Generate the default pod spec for the given control plane, if user has
+		// provided custom config for the etcd pod spec, patch this user
+		// provided config to the default spec
+		etcdSpec, err := patch.PodSpec(PodSpecFor(controlPlane), controlPlane.Spec.Etcd.Spec)
+		if err != nil {
+			return fmt.Errorf("failed to patch pod spec, %w", err)
+		}
+		statefulSet.Spec = appsv1.StatefulSetSpec{
```

Comment:
We should be able to put this spec in at line 171, but if it's not working for you for some reason, I can dig in later.